mp2-v1/src/values_extraction/api.rs

```diff
@@ -777,105 +870,50 @@ mod tests {
                 .unwrap();
             let vk = params.set.verifier_data_for_input_proofs::<1>()[0].clone();
             ProofWithVK::from((fake_proof[0].clone(), vk))
+                .serialize()
+                .unwrap()
         };
 
-        // Check validity of public input of `branch2` proof.
-        let check_public_input = |num_children, proof: &ProofWithVK| {
-            let branch_pub = PublicInputs::new(&proof.proof().public_inputs[..NUM_IO]);
-
-            let value: Vec<u8> = rlp::decode(&trie.get(mpt1).unwrap().unwrap()).unwrap();
-            let [leaf_values_digest, leaf_metadata_digest] = if is_simple_aggregation {
-                let dv = compute_leaf_single_values_digest(id, &value);
-                let dm = compute_leaf_single_metadata_digest(id, TEST_SLOT);
-
-                [dv, dm]
-            } else {
-                let dv = compute_leaf_mapping_values_digest(
-                    key_id,
-                    value_id,
-                    &test_data.mapping_key.clone().unwrap(),
-                    &value,
-                );
-                let dm = compute_leaf_mapping_metadata_digest(key_id, value_id, TEST_SLOT);
-
-                [dv, dm]
-            };
+        // Check the public input of branch proof.
+        let check_branch_public_inputs = |num_children, branch_proof: &ProofWithVK| {
+            let [leaf_pi, branch_pi] = [&leaf_proof1, branch_proof]
+                .map(|proof| PublicInputs::new(&proof.proof().public_inputs[..NUM_IO]));
 
-            let values_digest =
+            let leaf_metadata_digest = leaf_pi.metadata_digest();
```

Comment:
Maybe here we should check that the metadata digest corresponds to the one computed from `table_info`, as it was done in the previous version of the test (where we were recomputing metadata digest outside of circuit with `compute_leaf_mapping_metadata_digest`)? If I am not wrong, the `leaf_pi.metadata_digest` is never checked against the expected one, so checking that this digest is the same as the one of the leaf proof might be not so meaningful Wdyt?